### PR TITLE
loginUser func completion callback fix

### PR DESCRIPTION
**Pull request by mehmetdelikaya**
Hi @maurovz ,

Thanks a lot for the library. I was trying to implement something similar into my project and I have found out that you have implemented this library and easier to integrate rather than walletconnect itself.

I am using your library in my swift project and I have noticed that I don't get the callback to get wallet details after login when I use the snippet below,
`glaip.loginUser(type: .WalletConnect) { result in
  print("RESULT == \(result)")
}`

I have noticed that you don't call completion at success case, and also I had to change first parameter of return result string with User object to access user's wallet data. Later on, I have noticed I still don't get the callback. After my investigation, I had to remove async operation after wallet connect completion received. Everything works well on my side now, feel free to discuss. 

Cheers

### CODE REVIEW

File: Sources/Glaip/API/Glaip.swift [hunk 1]
```diff
@@ -27,35 +27,33 @@ public final class Glaip: ObservableObject {
...
 
-  public func loginUser(type: WalletType, completion: @escaping (Result<String, Error>) -> Void) {
+  public func loginUser(type: WalletType, completion: @escaping (Result<User, Error>) -> Void) {
     switch type {
     case .WalletConnect:
       metaMaskLogin(completion: { [weak self] result in
         guard let self = self else { return }
 
-        DispatchQueue.main.async {
           switch result {
```
Comment (maurovz) [inline Sources/Glaip/API/Glaip.swift:36]: We actually need to call the main thread for `self.userState = .loggedIn(user)` but I can address it in a separate PR

**Review by maurovz — APPROVED** [review]
Thanks for your contribution!!

File: Sources/Glaip/API/Glaip.swift (hunk 1 above)
Comment (mehmetdelikaya) [reply Sources/Glaip/API/Glaip.swift:36]: @maurovz Ok I see, I will test that in my case and I will let you know

Comment (mehmetdelikaya) [reply Sources/Glaip/API/Glaip.swift:36]: @maurovz I tested it and it still works, thanks bro

**Pull request merged**